src/k8s/pkg/k8sd/api/worker_access_handler.go

```diff
@@ -24,3 +28,38 @@ func RestrictWorkers(s *state.State, r *http.Request) response.Response {
 
 	return response.EmptySyncResponse
 }
+
+// TokenAuthentication access handler checks if the worker is allowed to access this endpoint with the provided token.
+// TokenAuthentication expects the "worker-name" and "worker-secret" fields to be set in the request header.
+func TokenAuthentication(s *state.State, r *http.Request) response.Response {
```

Comment:
Let's give it a name that describes the usage without having to lookup the implementation. Also, make this a closure that accepts the header names and returns the handler

```suggestion
func ValidateWorkerInfoAccessHandler(nodeHeaderName string, tokenHeaderName string) func(s *state.State, r *http.Request) response.Response {
```

then used as `AccessHandler: ValidateWorkerInfoAccessHandler("worker-name", "worker-token")`